test/commitment/proof_of_work.cpp

```diff
@@ -0,0 +1,140 @@
+        test_global_seed = 0;
+
+        for( std::size_t i = 0; i + 1 < boost::unit_test::framework::master_test_suite().argc; i++){
+            if(std::string(boost::unit_test::framework::master_test_suite().argv[i]) == "--seed"){
+                if(std::string(boost::unit_test::framework::master_test_suite().argv[i+1]) == "random"){
+                    std::random_device rd;
+                    test_global_seed = rd();
+                    std::cout << "Random seed=" << test_global_seed << std::endl;
+                    break;
+                }
+                if(std::regex_match( boost::unit_test::framework::master_test_suite().argv[i+1], std::regex( ( "((\\+|-)?[[:digit:]]+)(\\.(([[:digit:]]+)?))?" ) ) ) ){
+                    test_global_seed = atoi(boost::unit_test::framework::master_test_suite().argv[i+1]);
+                    break;
+                }
+            }
+        }
+
+        BOOST_TEST_MESSAGE("test_global_seed = " << test_global_seed);
+        test_global_rnd_engine = boost::random::mt11213b(test_global_seed);
+        test_global_alg_rnd_engine<field1_type> = nil::crypto3::random::algebraic_engine<field1_type>(test_global_seed);
+    }
+    ~test_fixture(){}
+};
+
+
+using namespace nil::crypto3::algebra;
+using namespace nil::actor::zk::commitments;
+
+ACTOR_FIXTURE_TEST_CASE(pow_poseidon_basic_test, test_fixture) {
+    using curve_type = curves::pallas;
+    using field_type = curve_type::base_field_type;
+    using integral_type = typename field_type::integral_type;
+    using policy = nil::crypto3::hashes::detail::mina_poseidon_policy<field_type>;
+    using poseidon = nil::crypto3::hashes::poseidon<policy>;
+    using pow_type = nil::actor::zk::commitments::field_proof_of_work<poseidon, field_type, 32>;
+
+    const integral_type expected_mask = integral_type(0xFFFFFFFF00000000) << (field_type::modulus_bits - 64);
+    nil::actor::zk::transcript::fiat_shamir_heuristic_sequential<poseidon> transcript;
+    auto old_transcript_1 = transcript, old_transcript_2 = transcript;
+
+    auto seed = nil::actor::testing::local_random_engine();
+    BOOST_TEST_MESSAGE("Seed for proof of work generator: " << seed);
+    nil::crypto3::random::algebraic_engine<field_type> rnd_engine = nil::crypto3::random::algebraic_engine<field_type>(seed);
+
+    auto result = pow_type::generate(transcript,  rnd_engine);
+    BOOST_ASSERT(expected_mask == pow_type::mask);
+    BOOST_ASSERT(pow_type::verify(old_transcript_1, result));
+
+    // manually reimplement verify to ensure that changes in implementation didn't break it
+    old_transcript_2(result);
+    auto chal = old_transcript_2.template challenge<field_type>();
+    BOOST_ASSERT((integral_type(chal.data) & expected_mask) == 0);
+
+    using hard_pow_type = nil::actor::zk::commitments::field_proof_of_work<poseidon, field_type, 32>;
+    // check that random stuff doesn't pass verify
+    BOOST_ASSERT(!hard_pow_type::verify(old_transcript_1, result));
+}
+/*
```

Comment:
Either delete this commented test, or just disable it with *boost::unit_test::disabled().

Also the upper test, I believe it's very slow, please disable it.